solr/packaging/powershell-tests/Zk.Tests.ps1

```diff
@@ -42,6 +42,81 @@ BeforeAll {
   $script:ZK_PORT = if ($env:ZK_PORT) { $env:ZK_PORT } else { "9983" }
   $script:SOLR_PORT = if ($env:SOLR_PORT) { $env:SOLR_PORT } else { "8983" }
 
+  # Check if Solr is already running
+  $solrAlreadyRunning = $false
+  try {
+    $response = Invoke-WebRequest -Uri "http://localhost:$SOLR_PORT/solr/" -UseBasicParsing -TimeoutSec 2 -ErrorAction SilentlyContinue
+    if ($response.StatusCode -eq 200) {
+      $solrAlreadyRunning = $true
+      Write-Host "Solr is already running on port $SOLR_PORT"
+    }
+  } catch {
+    $solrAlreadyRunning = $false
+  }
+
+  $script:SolrStartedByTests = $false
+
+  # Start Solr if it's not already running
+  if (-not $solrAlreadyRunning) {
+    Write-Host "Starting Solr in cloud mode..."
+
+    # Start Solr with embedded ZooKeeper using Start-Process to run asynchronously
+    $startArgs = @("start", "-p", "$SOLR_PORT")
+    Write-Host "Running: $SolrCmd $($startArgs -join ' ')"
+
+    # Use Start-Job with cmd.exe to properly parse arguments and maintain process hierarchy
+    $solrJob = Start-Job -ScriptBlock {
+        param($cmd, $argString)
+        cmd /c "`"$cmd`" $argString" 2>&1
+    } -ArgumentList $SolrCmd, $startArgs
+
+    # Wait a bit for Solr to start
+    Start-Sleep -Seconds 5
```

Comment:
Is this necessary? Our start script waits until solr is ready before `bin\solr.cmd start` exits?